go/engine/login_provision.go

```diff
@@ -469,6 +470,12 @@ func (e *loginProvision) makeDeviceKeys(ctx *Context, args *DeviceWrapArgs) erro
 
 	e.signingKey = eng.SigningKey()
 	e.encryptionKey = eng.EncryptionKey()
+
+	did := e.G().Env.GetDeviceID()
+
+	e.sharedDHKeyring = libkb.NewSharedDHKeyring(e.G(), e.arg.User.GetUID())
```

Comment:
Does it make sense to always require a `DeviceID` as part of the constructor?  It seems like it should to me.